shiny/ui/_sidebar.py

```diff
@@ -312,62 +349,76 @@ def _as_open(
         if open is None:
             return None
 
-        if isinstance(open, str):
-            return SidebarOpen._from_string(open)
-
-        if isinstance(open, dict):
-            return SidebarOpen(**open)
-
         if isinstance(open, SidebarOpen):
             return open
 
-        raise ValueError("`open` must be a string or a `SidebarOpen` object")
+        return SidebarOpen._as_open(open)
 
     def _set_default_open(
         self, open: SidebarOpen | SidebarOpenSpec | SidebarOpenValue
     ) -> Sidebar:
+        """
+        Sets the default or fallback value of `open` for the sidebar. In other words,
+        this determines the fallback value of `open` if the user did not set `open` when
+        they called `sidebar()`.
+
+        This method avoids modifying the original object in place and returns a new copy
+        of the sidebar object if needed.
+        """
         new_default = self._as_open(open)
 
         if new_default is None:
             raise ValueError(
                 "`open` must be a string, a `SidebarOpen` object, or a dictionary with keys `desktop` and `mobile`."
             )
 
+        if new_default == self._default_open:
+            return self
+
         new = copy(self)
         new._default_open = new_default
 
         return new
 
     def _resolved_sidebar_id(self) -> Optional[str]:
```

Comment:
Just wondering: Will this work in a module? I think this function should return a `ResolvedId` instead of `str`, but I'm not 100% sure.

Reply:
Yes. @schloerke and I talked about this quite a bit. We settled on resolving the `id` on initialization. Initially I had this method resolving the `id` when tagifying, which meant that you could re-use a sidebar in or out of a module, but it also causes issues if the object isn't tagified in the right session context.

Anyway, at this point, this method isn't as much about resolving the `id` as it is about returning either the user's ID or a randomly generated ID (if needed). That feels similar to _resolving the ID_, but could probably be renamed to avoid overlap with `ResolvedId`.

I'll just rename it `._get_sidebar_id()`.